cvat/apps/engine/model_utils.py

```diff
@@ -18,3 +22,86 @@ class Undefined:
 The reverse side of one-to-one relationship.
 May be undefined in the object, should be accessed via getattr().
 """
+
+
+_ModelT = TypeVar("_ModelT", bound=models.Model)
+
+
+def bulk_create(
+    db_model: type[_ModelT],
+    objs: Iterable[_ModelT],
+    batch_size: int | None = ...,
+    ignore_conflicts: bool = ...,
+    update_conflicts: bool | None = ...,
+    update_fields: Sequence[str] | None = ...,
+    unique_fields: Sequence[str] | None = ...,
```

Comment:
Can't this be replaced with `**kwargs`?

Reply:
It can, I just wanted it to be a "drop-in" replacement with the same signature. It can't be a pure drop-in, because now we require the first model argument, but this design offers good enough compatibility.

Reply:
Okay, but then why use the ellipses? Seems like you could copy the default values from the original signature as well.

Reply:
I used annotations from https://github.com/sbdchd/django-types/blob/062b1bfe5313b7e9f1a0a165fa5bf549e6b4addd/django-stubs/db/models/query.pyi#L52 that are [used by VS Code](https://github.com/microsoft/pylance-release/issues/4562#issuecomment-1615049922), which is our default development environment. The [more official django-stubs repo](https://github.com/typeddjango/django-stubs/blob/008c6e6b2534d9cf444a9650330611ca2248a4ee/django-stubs/db/models/query.pyi#L78) has the same annotations. Ellipses are supposed to mean "unchanged", and this is enforced by the implementation. If the official defaults ever change, we will not have to do any changes in this function.

Reply:
> and what most developers will typically see in the editor help.

I mean... would you rather see ellipses or actual values?

> The [more official django-stubs repo](https://github.com/typeddjango/django-stubs/blob/008c6e6b2534d9cf444a9650330611ca2248a4ee/django-stubs/db/models/query.pyi#L78) has the same annotations.

No it doesn't, check again. 🙂 

> Ellipses are supposed to mean "unchanged", and this is enforced by the implementation. If the official defaults ever change, we will not have to do any changes in this function.

I think the official defaults changing would imply a backwards compatibility break, so wouldn't it actually be useful for us to have hardcoded them?

Reply:
> No it doesn't, check again. 🙂

Ok, probably I missed this.

> I think the official defaults changing would imply a backwards compatibility break, so wouldn't it actually be useful for us to have hardcoded them?

I think it doesn't matter. Time spent on discussing this question is probably bigger than for changing the code in such case.

It seems Django is quite confident about the values, so I copied them from the "official" stub.